File: frontside/src/types/accountAnalysis.ts
export interface AccountChildBalance {
  account_id:   string;
  parent_id:    string | null;
  name:         string;
  is_summary:   boolean;
  has_child:    boolean;
  debit_total:  string;
  credit_total: string;
}

export interface AccountChildrenBalanceResult {
  account_id: string;
  name:       string;
  children:   AccountChildBalance[];
}

export interface AccountJournalEntryRow {
  entry_id:   number;
  txn_id:     number;
  account_id: string;
  ledger_id:  number | null;
  debit:      string;
  credit:     string;
  note:       string | null;
  txn_date:   string;
  description: string;
}

export interface AccountMonthlyBalance {
  month:        string;
  debit_total:  string;
  credit_total: string;
  has_snapshot: boolean;
}

export interface PagedMeta {
  page:        number;
  page_size:   number;
  total_pages: number;
  total:       number;
}

export interface PagedResponse<T> {
  data: T[];
  meta: PagedMeta;
}

File: backend/internal/model/db/projection/account_analysis.go
package projection

import "github.com/shopspring/decimal"

// AccountChildBalance is a composite type (manual mapping, no dbmap annotation).
type AccountChildBalance struct {
	AccountID     string          `db:"account_id" json:"account_id"`
	ParentID      *string         `db:"parent_id" json:"parent_id"`
	Name          string          `db:"name" json:"name"`
	IsSummary     bool            `db:"is_summary" json:"is_summary"`
	HasChild      bool            `db:"has_child" json:"has_child"`
	DebitTotal    decimal.Decimal `json:"debit_total"`
	CreditTotal   decimal.Decimal `json:"credit_total"`
}

type AccountChildrenBalanceResult struct {
	AccountID   string                `json:"account_id"`
	Name        string                `json:"name"`
	Children    []AccountChildBalance `json:"children"`
}

// AccountJournalEntryRow is a composite type (manual mapping, no dbmap annotation).
type AccountJournalEntryRow struct {
	EntryID     int64           `db:"entry_id" json:"entry_id"`
	TxnID       int64           `db:"txn_id" json:"txn_id"`
	AccountID   string          `db:"account_id" json:"account_id"`
	LedgerID    *int64          `db:"ledger_id" json:"ledger_id"`
	Debit       decimal.Decimal `db:"debit" json:"debit"`
	Credit      decimal.Decimal `db:"credit" json:"credit"`
	Note        *string         `db:"note" json:"note"`
	TxnDate     string          `db:"txn_date" json:"txn_date"`
	Description string          `db:"description" json:"description"`
}

// AccountMonthlyBalance is computed in Go from snapshot + delta queries.
type AccountMonthlyBalance struct {
	Month       string          `json:"month"`
	DebitTotal  decimal.Decimal `json:"debit_total"`
	CreditTotal decimal.Decimal `json:"credit_total"`
	HasSnapshot bool            `json:"has_snapshot"`
}
